packit_service/worker/handlers/centosmsg_handlers.py

```diff
@@ -0,0 +1,156 @@
+from packit_service.worker.handlers.abstract import use_for, JobHandler
+from packit_service.worker.result import HandlerResults
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractPagureJobHandler(JobHandler, PagurePackageConfigGetter):
+    pass
+
+
+class AbstractPagureCoprBuildHandler(AbstractPagureJobHandler):
+    event: Union[PullRequestPagureEvent]
+
+    def __init__(
+        self,
+        config: ServiceConfig,
+        job_config: JobConfig,
+        event: Union[PullRequestPagureEvent, PushPagureEvent],
+    ):
+        super().__init__(config=config, job_config=job_config, event=event)
+
+        # lazy property
+        self._copr_build_helper: Optional[CoprBuildJobHelper] = None
+        self._package_config: Optional[PackageConfig] = None
+        self._project: Optional[GitProject] = None
+
+    @property
+    def copr_build_helper(self) -> CoprBuildJobHelper:
+        if not self._copr_build_helper:
+            self._copr_build_helper = CoprBuildJobHelper(
+                config=self.config,
+                package_config=self.package_config,
+                project=self.project,
+                event=self.event,
+                job=self.job_config,
+            )
+        return self._copr_build_helper
+
+    @property
+    def project(self) -> GitProject:
+        if not self._project:
+            self._project = self.event.get_project()
+        return self._project
+
+    @property
+    def package_config(self) -> PackageConfig:
+        if not self._package_config:
+            self._package_config = self.event.get_package_config()
+            self._package_config.upstream_project_url = self.event.project_url
+        return self._package_config
+
+    def run(self) -> HandlerResults:
+        return self.copr_build_helper.run_copr_build()
+
+    def pre_check(self) -> bool:
+        is_copr_build: Callable[
+            [JobConfig], bool
+        ] = lambda job: job.type == JobType.copr_build
+
+        if self.job_config.type == JobType.tests and any(
+            filter(is_copr_build, self.package_config.jobs)
+        ):
+            logger.info(
+                "Skipping build for testing. The COPR build is defined in the config."
+            )
+            return False
+        return True
+
+
+@use_for(job_type=JobType.copr_build)
+@use_for(job_type=JobType.build)
+class PagurePullRequestCoprBuildHandler(AbstractPagureCoprBuildHandler):
+    triggers = [
+        TheJobTriggerType.pull_request,
+    ]
+    event: PullRequestPagureEvent
+
+    def __init__(
+        self,
+        config: ServiceConfig,
+        job_config: JobConfig,
+        event: PullRequestPagureEvent,
+    ):
+        super().__init__(config=config, job_config=job_config, event=event)
+
+    def run(self) -> HandlerResults:
+        collaborators = self.project.who_can_merge_pr()
+        if self.event.user_login not in collaborators | self.config.admins:
```

Comment:
I'd say that we don't need to check for this - we can build everything

Reply:
fixed